src/acktest/bootstrapping/s3.py

```diff
@@ -0,0 +1,48 @@
+import boto3
+import logging
+import json
+import re
+import time
+
+from dataclasses import dataclass, field
+from typing import List
+
+from . import BootstrappableResource
+from .. import resources
+from ..aws.identity import get_region, get_account_id
+
+@dataclass
+class Bucket(BootstrappableResource):
+    # Inputs
+    name_prefix: str
+
+    # Outputs
+    name: str = field(init=False)
+    
+    def bootstrap(self):
+        """Creates an S3 bucket with an auto-generated name.
+        """
+        region = get_region()
+        self.name = resources.random_suffix_name(self.name_prefix, 63)
```

Comment:
What if the caller wants to specify a particular name and not have it be randomized?

Reply:
I highly don't recommend that. That means you can bootstrap for at least one test per account, limiting you to run only a single integration test at a time. Would there be a reason to have a completely deterministic resource name?

Reply:
I don't disagree with you. I was asking whether it would be possible to give it a particular name instead of always be randomized.

Reply:
Yes it would be possible, but I pretty explicitly gated it so that it wouldn't be. We could easily change `name_prefix` to `name` and make it part of the constructor like any other field.